src/MigrationsCommandFactory.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\PsrContainerDoctrine;
+
+use Doctrine\Migrations\Tools\Console\Command\AbstractCommand;
+use Psr\Container\ContainerInterface;
+use function array_key_exists;
+use function class_exists;
+use function in_array;
+use function sprintf;
+use function str_replace;
+use function strtolower;
+use function ucfirst;
+
+class MigrationsCommandFactory
+{
+    /** @var string */
+    private $commandKey;
+
+    /** @var array<string> */
+    private static $availableCommands = [
+        'abstract',
```

Comment:
Probably best to have references to classes here, and not hardcoded strings: that would help static analysis in identifying this as a `list<class-string<Command>>`